cmake/Findhiredis.cmake

```diff
@@ -1,4 +1,38 @@
-if(HIREDIS_FROM_INTERNET)
+if(hiredis_FOUND)
+  return()
+endif()
+
+set(hiredis_FOUND FALSE)
+
+find_package(PkgConfig)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(HIREDIS hiredis>=${hiredis_FIND_VERSION})
+  find_library(HIREDIS_LIBRARY ${HIREDIS_LIBRARIES} HINTS ${HIREDIS_LIBDIR})
+  find_path(HIREDIS_INCLUDE_DIR hiredis/hiredis.h HINTS ${HIREDIS_PREFIX}/include)
+else()
+  find_library(HIREDIS_LIBRARY hiredis)
+  find_path(HIREDIS_INCLUDE_DIR hiredis/hiredis.h)
+endif()
+
+if(HIREDIS_INCLUDE_DIR AND HIREDIS_LIBRARY)
+  mark_as_advanced(HIREDIS_INCLUDE_DIR HIREDIS_LIBRARY)
+
+  add_library(HIREDIS::HIREDIS UNKNOWN IMPORTED)
+  set_target_properties(
+    HIREDIS::HIREDIS
+    PROPERTIES
+    IMPORTED_LOCATION "${HIREDIS_LIBRARY}"
+    INTERFACE_COMPILE_OPTIONS "${HIREDIS_CFLAGS_OTHER}"
+    INTERFACE_INCLUDE_DIRECTORIES "${HIREDIS_INCLUDE_DIR}")
+  if(WIN32 AND STATIC_LINK)
+    target_link_libraries(HIREDIS::HIREDIS INTERFACE ws2_32)
+  endif()
+
+  set(hiredis_FOUND TRUE)
+  set(target HIREDIS::HIREDIS)
+elseif(HIREDIS_FROM_INTERNET)
+  message(STATUS "*** WARNING ***: Using hiredis from the internet because it was NOT found and HIREDIS_FROM_INTERNET is TRUE")
```

Comment:
Would it make sense to use `message(WARNING [...])` here (and similar for zstd)?

Reply:
CMake warnings are very ugly, they are intended more for warnings about cmake code and show the line number etc.. Here I wanted to show a loud warning that also blends into the rest of the output, but I will change this if you insist.

Reply:
OK, then let's keep it as is. :+1: